plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -406,12 +410,17 @@ struct state_history_plugin_impl : std::enable_shared_from_this<state_history_pl
 
       std::vector<table_delta> deltas     = state_history::create_deltas(chain_plug->chain().db(), fresh);
       auto                     deltas_bin = state_history::zlib_compress_bytes(fc::raw::pack(deltas));
-      EOS_ASSERT(deltas_bin.size() == (uint32_t)deltas_bin.size(), plugin_exception, "deltas is too big");
       state_history_log_header header{.magic        = ship_magic(ship_current_version),
                                       .block_id     = block_state->block->id(),
                                       .payload_size = sizeof(uint32_t) + deltas_bin.size()};
       chain_state_log->write_entry(header, block_state->block->previous, [&](auto& stream) {
+         // Compressed deltas now exceeds 4GB on one of the public chains. This length prefix
+         // was intended to support adding additional fields in the future after the
+         // packed deltas. For now we're going to ignore on read. The 0 is an attempt to signal
+         // old versions that something's not quite right.
```

Comment:
They try to uncompress an empty buffer. I don't know off-hand whether boost iostreams will throw because of a missing header or if it will return empty data. The client will get either a disconnect or an empty result stream.